disperser/dataapi/server_v2.go

```diff
@@ -245,16 +251,96 @@ func (s *ServerV2) FetchBatchHandler(c *gin.Context) {
 	c.JSON(http.StatusOK, batchResponse)
 }
 
+// FetchOperatorsStake godoc
+//
+//	@Summary	Operator stake distribution query
+//	@Tags		OperatorsStake
+//	@Produce	json
+//	@Param		operator_id	query		string	false	"Operator ID in hex string [default: all operators if unspecified]"
+//	@Success	200			{object}	OperatorsStakeResponse
+//	@Failure	400			{object}	ErrorResponse	"error: Bad request"
+//	@Failure	404			{object}	ErrorResponse	"error: Not found"
+//	@Failure	500			{object}	ErrorResponse	"error: Server error"
+//	@Router		/operators/stake [get]
 func (s *ServerV2) FetchOperatorsStake(c *gin.Context) {
-	errorResponse(c, errors.New("FetchOperatorsStake unimplemented"))
+	timer := prometheus.NewTimer(prometheus.ObserverFunc(func(f float64) {
+		s.metrics.ObserveLatency("FetchOperatorsStake", f*1000) // make milliseconds
```

Comment:
These can be replaced with grpc interceptor?

Reply:
Oh wait dataapi is http, but I think similar concept might apply?

Reply:
https://github.com/Depado/ginprom

Reply:
It looks a hobby project, otherwise we could drop these primitive level metrics in application layer.